includes/log-handlers/class-wc-log-handler-email.php

```diff
@@ -0,0 +1,135 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * Handles log entries by sending an email.
+ *
+ * @class          WC_Log_Handler_Email
+ * @version        1.0.0
+ * @package        WooCommerce/Classes/Log_Handlers
+ * @category       Class
+ * @author         WooThemes
+ */
+class WC_Log_Handler_Email extends WC_Log_Handler {
+
+	/**
+	 * Minimum log level this handler will process.
+	 *
+	 * @var int Integer representation of minimum log level to handle.
+	 * @access private
+	 */
+	protected $threshold;
+
+	/**
+	 * Stores email recipients.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	private $recipients = array();
+
+	/**
+	 * Constructor for log handler.
+	 *
+	 * @param string|array $recipients Optional. Email(s) to receive log messages. Defaults to site admin email.
+	 * @param string $threshold Optional. Minimum level that should receive log messages.
+	 *     Default 'alert'. One of: emergency|alert|critical|error|warning|notice|info|debug
+	 */
+	public function __construct( $recipients = null, $threshold = 'alert' ) {
+		if ( null === $recipients ) {
+			$recipients = get_option( 'admin_email' );
+		}
+
+		if ( is_array( $recipients ) ) {
+			foreach ( $recipients as $recipient ) {
+				$this->add_email( $recipient );
+			}
+		} else {
+			$this->add_email( $recipients );
+		}
+
+		$this->set_threshold( $threshold );
+	}
+
+	/**
+	 * Set handler severity threshold.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 */
+	public function set_threshold( $level ) {
+		$this->threshold = WC_Log_Levels::get_level_severity( $level );
+	}
+
+	/**
+	 * Determine whether handler should handle log.
+	 *
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @return bool True if the log should be handled.
+	 */
+	public function should_handle( $level ) {
+		return $this->threshold <= WC_Log_Levels::get_level_severity( $level );
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message Log message.
+	 * @param array $context Optional. Additional information for log handlers.
+	 *
+	 * @return bool True on success.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
+
+		if ( $this->should_handle( $level ) ) {
+			$subject = $this->get_subject( $timestamp, $level, $message, $context );
+			$body = $this->get_body( $timestamp, $level, $message, $context );
+			return wp_mail( $this->recipients, $subject, $body );
+		}
+
+		return true;
```

Comment:
The return signature here is `@return bool True on success`, so `true` and `false` represent success and failure, respectively. The `true` in this case says essentially represents the success of no-op. We decided not to handle the log, and were successful in that. This is confusing if you interpret the return signature as "I did something with the log".

From a practical standpoint, the return value is [currently ignored by the logger](https://github.com/woocommerce/woocommerce/blob/54d131da39b00469ca104dcff828e830546aae25/includes/class-wc-logger.php#L110-L112). In the case that the logger decides to use the value in the future, it's hard to anticipate whether it's better to indicate "success/failure" or "handled/not handled (maybe failed?)". It's also hard to represent these nuances with the binary `true`/`false`.

It may be best to remove the return value entirely and add it when and if we have a clear use case in the future. What are your thoughts?